applications/DigitalTwinApplication/custom_sensors/displacement_sensor.cpp

```diff
@@ -0,0 +1,254 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         DigitalTwinApplication/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System includes
+#include <cmath>
+
+// External includes
+
+// Project includes
+
+// Application includes
+#include "digital_twin_application_variables.h"
+
+// Include base h
+#include "displacement_sensor.h"
+
+namespace Kratos {
+
+/// Constructor.
+DisplacementSensor::DisplacementSensor(
+    const std::string& rName,
+    const Point& rLocation,
+    const array_1d<double, 3>& rDirection,
+    const Element& rElement,
+    const double Weight)
+    : BaseType(rName, rLocation, Weight),
+      mElementId(rElement.Id()),
```

Comment:
Oo
when does reordering happen in MPI?

Reply:
at reading time. Metis requires consecutive ordering, for which we use IDs. Technically this can be solved with a Map, but still my other issue with Ids is that it is a search when accessing entities by Id!

Reply:
Well, that's pretty bad. An ID is supposed to be stable and immutable (which is why they're called IDs in the first place). If we cannot rely on stable IDs, we can't use `PointerVectorSet` and there's no way for us to refer to specific elements/conditions/nodes/geometries in JSON (or any other data that isn't currently loaded in memory).

Reply:
I put the entities I am interested in into separate submodelparts

Worked flawless for me for years, not sure why everyone insists on using Ids 🤷🏻‍♂️

Reply:
For the time being, I have to rely on the Id. I will think of a proper solution. This ID issue needs to be adddressed anyways :)